great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -309,6 +313,39 @@ def run(
 
         return expectation_suite
 
+    def reconcile_batch_requests_in_builders(
+        self, batch_request: Union[dict, BatchRequest, RuntimeBatchRequest]
+    ) -> None:
```

Comment:
Should this be public? Is there a use-case outside of the `run_profiler_on_data` method?

Reply:
@cdkini It seems to me that this method should be private for several reason, one jumping into my mind is being that it makes several simplifying assumptions, which make it less general than the method it rests on.  Thanks!